### ISSUE TITLE
Bootstraper doesn't auto-update the launcher when an update is pushed.

### ISSUE THREAD

**Issue by halvors** [SKCraft/Launcher#55]
Bootstraper doesn't auto-update the launcher when an update is pushed. I have to manually click "Update Launcher" next to the "Launch" button.

It's preferable that the bootstraper looks for updates everytime you run the launcher, that wouldn't require the user to manually click the "Update Launcher" button.

**Comment by sk89q** [comment]
It does look for updates when the launcher starts (otherwise you wouldn't see the button) but your're right in that it doesn't auto-update.

**Comment by halvors** [comment]
Could that be added as an option? Some less experienced users may avoid updating then. And this is a problem when changing major things like CDN.

**Comment by sk89q** [comment]
It's doable but I haven't yet because I haven't implemented background tasks.

**Comment by halvors** [comment]
I meant on startup, exactly like it does the first time the bootstraper starts :+1:

**Comment by sk89q** [comment]
At the moment, the update check occurs silently in the background in the launcher (and not the bootstrapper), and the user may have tried to start a mod pack already by the time the update check completes. Do you want the update check to prevent the launcher from starting up so that this problem doesn't happen?

**Comment by dscso** [comment]
Whats the status of this issue? I personally prefer if the Bootstrapper automaticly downloads the new launcher update without asking (like the official Minecraft Bootstrapper does).

**Comment by sk89q** [comment]
That can be added pretty easily. If you want that, I can add that as an option. I don't like it when programs wait for an update before I get to use it, but it would be up to you.

A silent background updater, the original suggestion, is a more involved.

**Comment by dscso** [comment]
Yes. A silent background updater is the better way. It installs the new jar and asks if it should restart.
